src/plugins/drawing-tool/model/drawing-content.test.ts

```diff
@@ -121,55 +135,89 @@ describe("DrawingContentModel", () => {
       strokeWidth: DefaultToolbarSettings.strokeWidth
     };
     expect(model.toolbarSettings).toEqual(defaultSettings);
-    model.setStroke(stroke);
+    model.setStroke(stroke, model.selectedIds);
     expect(model.toolbarSettings).toEqual({ ...defaultSettings, stroke });
-    model.setFill(fill);
+    model.setFill(fill, model.selectedIds);
     expect(model.toolbarSettings).toEqual({ ...defaultSettings, fill, stroke });
-    model.setStrokeDashArray(strokeDashArray);
+    model.setStrokeDashArray(strokeDashArray, model.selectedIds);
     expect(model.toolbarSettings).toEqual({ ...defaultSettings, fill, stroke, strokeDashArray });
-    model.setStrokeWidth(strokeWidth);
+    model.setStrokeWidth(strokeWidth, model.selectedIds);
     expect(model.toolbarSettings).toEqual({ ...defaultSettings, fill, stroke, strokeDashArray, strokeWidth });
   });
 
   it("can delete a set of selected drawing objects", () => {
     const model = createDrawingContentWithMetadata();
-    const {stroke, fill, strokeWidth, strokeDashArray} = mockSettings;
 
-    model.addObject(RectangleObject.create(
-      {id:"a", x:0, y:0, width:10, height:10,
-       stroke, fill, strokeWidth, strokeDashArray}));
+    // TODO: maybe we want to just reset the one function, but TS doesn't like us calling resetMock
+    // on Logger.logToolChange
+    jest.resetAllMocks();
```

Comment:
I suspect if you did something like:
```typescript
// mock Logger calls
var logToolChange = jest.fn();
jest.mock("../../../lib/logger", () => {
  return {
    ...(jest.requireActual("../../../lib/logger") as any),
    Logger: {
      logToolChange
    }
  };
});
```
up above then you'd be able to just:
```typescript
    logToolChange.resetMock();
```

Reply:
I did something slightly different because the Jest compiler moves jest.mock above all else in the file. I did use a `const` instead of a `var`, perhaps with a `var` I could have done it your way.

Reply:
Yes, the `var` is the key because it's hoisted as well. The eslint rules specifically allow `var` in test files but not regular source files for this reason:
```javascript
      { // test files
        files: ["*.test.*", "jest-resolver.js"],
        env: {
          node: true,
          jest: true
        },
        rules: {
          "@typescript-eslint/no-non-null-assertion": "off",
          // require() can be useful in mocking
          "@typescript-eslint/no-require-imports": "off",
          "@typescript-eslint/no-var-requires": "off",
          // var is useful in mocking due to its hoisting semantics
          "no-var": "off"
        }
      },
```